### PR TITLE
Added Babel Presets

### LINKED ISSUE #23: Dynamic imports result in "Module build failed: SyntaxError: Unexpected token"
Using a dynamic import such as:

`import('./path/to/file')`

Results in error:

    ERROR in ./app/src/routes.js
    Module build failed: SyntaxError: Unexpected token (15:9)

      13 | 	path: '**',
      14 | 	component: RouteUtils.asyncComponent(() => {
    > 15 | 		return import('./pods/site/pages/NotFoundPage');
         | 		       ^
      16 | 	}),
      17 | });

**Suggested solution:**

Pre-install https://babeljs.io/docs/plugins/syntax-dynamic-import/ as I would expect this functionality out of the box.

### PR DESCRIPTION
**Pull request by jennasalau**
I have added babel presets to manage our transforms.

This also resolves #23  

This includes decorators too as I want to start using them more. We have them at my current client site now and it makes the code so neat. Open to discussion on this one.

### CODE REVIEW

File: packages/demo/package.json [hunk 1]
```diff
@@ -7,15 +7,19 @@
...
     ]
   },
   "browserslist": [
     "last 2 versions",
     "IE 11"
   ],
   "devDependencies": {
-    "@deloitte-digital-au/webpack-config": "^0.0.5"
+    "@deloitte-digital-au/webpack-config": "^0.0.5",
+    "@deloitte-digital-au/babel-preset": "^0.0.5"
```
Comment (jennasalau) [inline packages/demo/package.json]: This is a mistake, I will fix but dont need to include babel-preset here as it comes with webpack-config.

File: packages/webpack-config/package.json [hunk 2]
```diff
@@ -12,10 +12,8 @@
   "dependencies": {
     "@deloitte-digital-au/eslint-config": "^3.4.0",
     "@deloitte-digital-au/stylelint-config": "^1.1.0",
+    "@deloitte-digital-au/babel-preset": "^0.0.5",
     "autoprefixer": "^8.3.0",
-    "babel-core": "^6.26.2",
-    "babel-loader": "^7.1.4",
```
Comment (jennasalau) [inline packages/webpack-config/package.json]: This is a mistake too, We should leave babel-loader here.

**Comment by jennasalau** [comment]
There is an issue with this branch where babel is expecting the presets to be installed locally but lerna is hoisting them up to the root.

I will need to think about how I resolve that later.

**Comment by keeganstreet** [comment]
I love the idea of the babel presets. But with decorators, I'm unsure about baking a stage 2 TC39 proposal into our standards. We will need to be prepared to update this as the proposal is refined. Are you finding decorators add any overhead to onboarding new developers who may be experienced with JavaScript but not with features that are not yet in the spec?

**Comment by jennasalau** [comment]
Yeah when you put it like that it doesn't make sense to have decorators in the core.. I will remove since its easy to add to a project manually if needed.

The team seem to be using them just fine, but granted I am the only one creating them so far.

**Comment by jennasalau** [comment]
@lewisvatcky can you please cast your eyes over this one and give it a review.

**Comment by lewisvatcky** [comment]
@jennasalau Sorry, didn't see this message! Looks good to me, excited to use decorators (finally).

**Comment by jennasalau** [comment]
All good. We pulled decorators out of this PR in the end so will need to install that manually still. Perhaps we can add some documentation on how to do that though.

**Pull request merged**